Library/Homebrew/cask/lib/hbc/installer.rb

```diff
@@ -367,12 +368,34 @@ def save_caskfile
     def uninstall
       oh1 "Uninstalling Cask #{@cask}"
       disable_accessibility_access
-      uninstall_artifacts
+      uninstall_artifacts(clear: true)
       purge_versioned_files
       purge_caskroom_path if force?
     end
 
-    def uninstall_artifacts
+    def start_upgrade
+      return unless upgrade?
```

Comment:
Remove this, this shouldn't be called anyway if this isn't an upgrade. Same goes for the methods below.